buildstockbatch/gcp/gcp.py

```diff
@@ -382,6 +382,28 @@ def clean(self):
         delete_job(self.gcp_batch_job_name)
         self.clean_postprocessing_job()
 
+        # Clean up images in Artifact Registry
+        ar_client = artifactregistry_v1.ArtifactRegistryClient()
+        repository = f"projects/{self.gcp_project}/locations/{self.region}/repositories/{self.ar_repo}"
+        package = f"{repository}/packages/buildstockbatch"
+        # Delete the tag used by this job
+        try:
+            ar_client.delete_tag(name=f"{package}/tags/{self.job_identifier}")
+        except exceptions.NotFound:
+            pass
+
+        # Then delete all untagged versions
+        all_versions = ar_client.list_versions(
+            artifactregistry_v1.ListVersionsRequest(parent=package, view=artifactregistry_v1.VersionView.FULL)
+        )
+        deleted = 0
+        for version in all_versions:
+            if not version.related_tags:
+                logger.debug(f"Deleting image {version.name}")
```

Comment:
nit/opinion: deleting is a significant enough operation worth of logging at info

Reply:
I think this is covered by logging the count of deleted images below.

The full version names aren't that user friendly or helpful most of the time, since they're indentified by their hashes. (e.g. `projects/buildstockbatch-dev/locations/us-central1/repositories/buildstockbatch-docker-test/packages/buildstockbatch/versions/sha256:a70b64025f9845aef053fc228b26bc7d39c1db2922b32daa34d45fbbba29640f`)